cluster/kind/kind.go

```diff
@@ -0,0 +1,172 @@
+	}
+	return string(b), nil
+}
+
+func clusterKindName() (string, error) {
+	name, err := clusterName()
+	if err != nil {
+		return "", fmt.Errorf("unable to determine cluster name: %v", err)
+	}
+	matches := clusterKindNameRE.FindStringSubmatch(name)
+	if len(matches) != 2 || matches[1] == "" {
+		return "", fmt.Errorf("cluster %v is not a kind cluster", name)
+	}
+	return matches[1], nil
+}
+
+func clusterKindNodes() ([]string, error) {
+	name, err := clusterKindName()
+	if err != nil {
+		return nil, err
+	}
+	out, err := run.OutCommand("kind", "get", "nodes", "--name", name)
+	if err != nil {
+		return nil, err
+	}
+	nodes := []string{}
+	for _, n := range strings.Split(string(out), " ") {
+		nodes = append(nodes, strings.TrimSuffix(n, "\n"))
+	}
+	return nodes, nil
+}
+
+func SetupGARAccess(ctx context.Context, registries []string) error {
+	// Create a temporary dir to hold a new docker config that lacks credsStore.
+	// Then use `docker login` to store the generated credentials directly in
+	// the temporary docker config.
+	// See https://kind.sigs.k8s.io/docs/user/private-registries/#use-an-access-token
+	// for more information.
+	tempDockerDir, err := os.MkdirTemp("", "kne_kind_docker")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tempDockerDir)
+	originalConfig := os.Getenv(dockerConfigEnvVar)
+	defer os.Setenv(dockerConfigEnvVar, originalConfig)
+	if err := os.Setenv(dockerConfigEnvVar, tempDockerDir); err != nil {
+		return err
+	}
+	configPath := filepath.Join(tempDockerDir, "config.json")
+	if err := writeDockerConfig(configPath, registries); err != nil {
+		return err
+	}
+	creds, err := googleFindDefaultCredentials(ctx, "https://www.googleapis.com/auth/cloud-platform")
+	if err != nil {
+		return fmt.Errorf("failed to find gcloud credentials: %v", err)
+	}
+	token, err := creds.TokenSource.Token()
+	if err != nil {
+		return fmt.Errorf("failed to get token from gcloud credentials: %v", err)
+	}
+	for _, r := range registries {
+		s := fmt.Sprintf("https://%s", r)
+		if err := run.LogCommandWithInput([]byte(token.AccessToken), "docker", "login", "-u", "oauth2accesstoken", "--password-stdin", s); err != nil {
+			return err
+		}
+	}
+	// Copy the new docker config to each node and restart kubelet so it
+	// picks up the new config that contains the embedded credentials.
+	nodes, err := clusterKindNodes()
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes {
+		if err := run.LogCommand("docker", "cp", configPath, fmt.Sprintf(kubeletConfigPathTemplate, node)); err != nil {
+			return err
+		}
+		if err := run.LogCommand("docker", "exec", node, "systemctl", "restart", "kubelet.service"); err != nil {
+			return err
+		}
+	}
+	log.Infof("Setup GAR access for %v", registries)
+	return nil
+}
+
+func RefreshGARAccess(ctx context.Context) error {
+	nodes, err := clusterKindNodes()
+	if err != nil {
+		return err
+	}
+	cfg, err := run.OutCommand("docker", "exec", nodes[0], "cat", kubeletConfigPath)
```

Comment:
Is it because there is only one node? or we are always interested in the first node returned?

Reply:
if one node doesnt have it, then none of them will, added a check and test case for no error but 0 nodes so thanks for pointing it out